topics/integrate-in-existing-app.md

```diff
@@ -1,242 +1,423 @@
+
+    ```kotlin
+    package com.jetbrains.simplelogin.shared
+    
+    import platform.Foundation.NSUUID
+    actual fun randomUUID(): String = NSUUID().UUIDString()
+    ```     
+
+For Android and iOS, Kotlin will use different platform-specific implementations.
+
+#### Run your cross-platform application
+
+Run your cross-platform application for Android to make sure that it works.
+
+![Android login application](android-login.png){width=300}
+
+## Make your cross-platform application work on iOS
+
+Once you've made your Android application cross platform, you can create an iOS application and reuse the shared business logic in it.
+
+1. [Create an iOS project in Xcode](#create-an-ios-project-in-xcode).
+
+2. [Compile the shared module into a framework for the iOS project](#compile-the-shared-module-into-a-framework-for-the-ios-project).
+
+3. [Connect the framework to your iOS project](#connect-the-framework-to-your-ios-project).
 
-### Create a KMM shared module
+4. [Automate iOS project updates](#automate-ios-project-updates).
 
-In your Android project, create a KMM shared module for your shared code.
+5. [Connect the shared module to the iOS project](#connect-the-shared-module-to-the-ios-project).
 
-> The KMM Shared Module Wizard is available in Android Studio version 4.1 or higher.
+
+### Create an iOS project in Xcode
+
+1. In Xcode, click **File** | **New** | **Project**.
+
+2. Select a template for an iOS app and click **Next**.
+
+    ![iOS project template](ios-project-wizard-1.png)
+
+3. As the product name, specify **simpleLoginIOS** and click **Next**.
+
+   ![iOS project settings](ios-project-wizard-2.png)
+
+4. As the location for your project, select the directory storing your cross-platform application, for example, `kmm-integrate-into-existing-app`.
+
+In Android Studio, you'll get the following structure:
+
+![iOS project in Android Studio](ios-project-in-as.png){width=194}
+
+You can rename the `simpleLoginIOS` directory to the `iosApp` directory for consistency with other top-level directories of your cross-platform project.
+
+![Renamed iOS project directory in Android Studio](ios-directory-renamed-in-as.png){width=194}
+
+### Compile the shared module into a framework for the iOS project
+
+To use Kotlin code in your iOS project, compile shared code into a `.framework`.
+
+* In Android Studio, run the `packForXcode` Gradle task in one of the following ways:
+
+   * Run the following command in the Terminal:
+
+       ```text
+        ./gradlew packForXcode
+       ```
+
+   * Double-click the `packForXcode` task in the list of Gradle tasks.
+   
+The generated framework is stored in `shared/build/xcode-frameworks/`.
+
+### Connect the framework to your iOS project
+
+Once you have the framework, you can connect it to your iOS project in different ways.
+
+This tutorial assumes that you store code for Android and iOS applications in a single repository (monorepo). 
```

Comment:
I believe this info should be earlier.

And here you can repeat it in that way:  **Since the tutorial assumes ...**